.github-ci.yml

```diff
@@ -0,0 +1,239 @@
+        cp -r _build/default/_doc/_html $ARTIFACTS/doc;
+        echo "Documentation:
+              $CI_JOB_URL/artifacts/browse/$ARTIFACTS/doc/index.html";
+      fi
+    - >
+      if [ -f _build/log ];
+      then
+        mkdir -p $ARTIFACTS/_build;
+        cp _build/log $ARTIFACTS/_build/log
+        echo "Dune build log:
+              $CI_JOB_URL/artifacts/browse/$ARTIFACTS/_build/log";
+      fi
+    - echo -e "section_end:`date +%s`:artifacts\r\e[0K"
+### CI performance
+
+# Performance analysis of one 'build' job on a small, simple library project
+# (the project build/test/doc time is basically neglectible) on gitlab.com:
+#
+# (To get those numbers for your project, just look at a passing build and fold all
+#  log sections.)
+#
+# 2m total
+#   56s preparing the docker image and CI environment
+#   14s restoring the _opam cache
+#   10s installing the project dependencies
+#       (external dependencies are not cached, opam dependencies are cached)
+#    3s project build, tests, doc
+#   31s saving the _opam cache
+#    6s uploading artifacts
+#
+# When running with CLEAN_OPAM_CACHE=true, the costs are similar, except for
+#  7m30s building the local opam switch
+#  2m07s building the project dependencies (external and opam)
+# This shows that caching the local _opam switch is very important, as
+# its setup time dominates the build (for a small project).
+#
+# Another thing that is clear from these numbers is that splitting
+# your CI in several jobs could be fairly expensive: for each job you
+# pay 56s seconds of docker setup, plus 14s to restore the _opam
+# cache. (The 31s to save the cache afterwards can be avoided by using
+# a "policy: pull" setting if the job uses the _opam in a read-only
+# way.)
+#
+# This is why this CI script does everything in a single 'build' step.
+
+## MIT License
+#
+# Copyright 2021 Gabriel Scherer
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files
+# (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge,
+# publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
```

Comment:
I think it passes on gitlab, but not on github, I had the same problem on another project in ocaml.

Reply:
If you want to, maybe you can look at that link: https://github.com/ThomArr/Chess/blob/main/.github/workflows/ocaml.yml

Reply:
I think at least you’re going to have to add: .github/workflows